teamcity: failed tests on master: testrace/TestWedgedReplicaDetection, testrace/TestWedgedReplicaDetection

The following tests appear to have failed:

[#426828](https://teamcity.cockroachdb.com/viewLog.html?buildId=426828):


```
--- FAIL: testrace/TestWedgedReplicaDetection (0.560s)
client_raft_test.go:1974: expected follower to be considered inactive
------- Stdout: -------
I171201 16:16:01.976126 52461 storage/store.go:1198  [s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I171201 16:16:01.976566 52461 gossip/gossip.go:331  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:34229" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
W171201 16:16:02.036014 52461 gossip/gossip.go:1277  [n2] no incoming or outgoing connections
I171201 16:16:02.044746 52674 gossip/client.go:129  [n2] started gossip client to 127.0.0.1:34229
I171201 16:16:02.057147 52461 storage/store.go:1198  [s2] [n2,s2]: failed initial metrics computation: [n2,s2]: system config not yet available
I171201 16:16:02.063380 52461 gossip/gossip.go:331  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:43273" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
W171201 16:16:02.168216 52461 gossip/gossip.go:1277  [n3] no incoming or outgoing connections
I171201 16:16:02.171202 52687 gossip/client.go:129  [n3] started gossip client to 127.0.0.1:34229
I171201 16:16:02.187063 52461 gossip/gossip.go:331  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:45179" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
I171201 16:16:02.213118 52862 storage/replica_raftstorage.go:527  [s1,r1/1:/M{in-ax}] generated preemptive snapshot ceb756bc at index 15
I171201 16:16:02.222839 52862 storage/store.go:3548  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):?: kv pairs: 34, log entries: 5, rate-limit: 2.0 MiB/sec, 9ms
I171201 16:16:02.224151 52880 storage/replica_raftstorage.go:733  [s2,r1/?:{-}] applying preemptive snapshot at index 15 (id=ceb756bc, encoded size=4722, 1 rocksdb batches, 5 log entries)
I171201 16:16:02.226128 52880 storage/replica_raftstorage.go:739  [s2,r1/?:/M{in-ax}] applied preemptive snapshot in 2ms [clear=0ms batch=0ms entries=1ms commit=0ms]
I171201 16:16:02.230706 52862 storage/replica_command.go:2153  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n2,s2):2): read existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2]
I171201 16:16:02.239256 52884 storage/replica.go:3158  [s1,r1/1:/M{in-ax}] proposing ADD_REPLICA((n2,s2):2): updated=[(n1,s1):1 (n2,s2):2] next=3
I171201 16:16:02.250141 52494 storage/replica_raftstorage.go:527  [s1,r1/1:/M{in-ax}] generated preemptive snapshot 6fc4538e at index 17
I171201 16:16:02.254210 52494 storage/store.go:3548  [s1,r1/1:/M{in-ax}] streamed snapshot to (n3,s3):?: kv pairs: 37, log entries: 7, rate-limit: 2.0 MiB/sec, 3ms
I171201 16:16:02.256178 52496 storage/replica_raftstorage.go:733  [s3,r1/?:{-}] applying preemptive snapshot at index 17 (id=6fc4538e, encoded size=5751, 1 rocksdb batches, 7 log entries)
I171201 16:16:02.258930 52496 storage/replica_raftstorage.go:739  [s3,r1/?:/M{in-ax}] applied preemptive snapshot in 3ms [clear=0ms batch=0ms entries=1ms commit=0ms]
I171201 16:16:02.262417 52494 storage/replica_command.go:2153  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n3,s3):3): read existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, next=3]
I171201 16:16:02.270053 52914 storage/raft_transport.go:455  raft transport stream to node 1 established
I171201 16:16:02.287873 52918 storage/replica.go:3158  [s1,r1/1:/M{in-ax}] proposing ADD_REPLICA((n3,s3):3): updated=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
--- FAIL: testrace/TestWedgedReplicaDetection (0.800s)
client_raft_test.go:1974: expected follower to be considered inactive
------- Stdout: -------
I171201 16:16:01.976126 52461 storage/store.go:1198  [s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I171201 16:16:01.976566 52461 gossip/gossip.go:331  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:34229" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
W171201 16:16:02.036014 52461 gossip/gossip.go:1277  [n2] no incoming or outgoing connections
I171201 16:16:02.044746 52674 gossip/client.go:129  [n2] started gossip client to 127.0.0.1:34229
I171201 16:16:02.057147 52461 storage/store.go:1198  [s2] [n2,s2]: failed initial metrics computation: [n2,s2]: system config not yet available
I171201 16:16:02.063380 52461 gossip/gossip.go:331  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:43273" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
W171201 16:16:02.168216 52461 gossip/gossip.go:1277  [n3] no incoming or outgoing connections
I171201 16:16:02.171202 52687 gossip/client.go:129  [n3] started gossip client to 127.0.0.1:34229
I171201 16:16:02.187063 52461 gossip/gossip.go:331  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:45179" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
I171201 16:16:02.213118 52862 storage/replica_raftstorage.go:527  [s1,r1/1:/M{in-ax}] generated preemptive snapshot ceb756bc at index 15
I171201 16:16:02.222839 52862 storage/store.go:3548  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):?: kv pairs: 34, log entries: 5, rate-limit: 2.0 MiB/sec, 9ms
I171201 16:16:02.224151 52880 storage/replica_raftstorage.go:733  [s2,r1/?:{-}] applying preemptive snapshot at index 15 (id=ceb756bc, encoded size=4722, 1 rocksdb batches, 5 log entries)
I171201 16:16:02.226128 52880 storage/replica_raftstorage.go:739  [s2,r1/?:/M{in-ax}] applied preemptive snapshot in 2ms [clear=0ms batch=0ms entries=1ms commit=0ms]
I171201 16:16:02.230706 52862 storage/replica_command.go:2153  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n2,s2):2): read existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2]
I171201 16:16:02.239256 52884 storage/replica.go:3158  [s1,r1/1:/M{in-ax}] proposing ADD_REPLICA((n2,s2):2): updated=[(n1,s1):1 (n2,s2):2] next=3
I171201 16:16:02.250141 52494 storage/replica_raftstorage.go:527  [s1,r1/1:/M{in-ax}] generated preemptive snapshot 6fc4538e at index 17
I171201 16:16:02.254210 52494 storage/store.go:3548  [s1,r1/1:/M{in-ax}] streamed snapshot to (n3,s3):?: kv pairs: 37, log entries: 7, rate-limit: 2.0 MiB/sec, 3ms
I171201 16:16:02.256178 52496 storage/replica_raftstorage.go:733  [s3,r1/?:{-}] applying preemptive snapshot at index 17 (id=6fc4538e, encoded size=5751, 1 rocksdb batches, 7 log entries)
I171201 16:16:02.258930 52496 storage/replica_raftstorage.go:739  [s3,r1/?:/M{in-ax}] applied preemptive snapshot in 3ms [clear=0ms batch=0ms entries=1ms commit=0ms]
I171201 16:16:02.262417 52494 storage/replica_command.go:2153  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n3,s3):3): read existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, next=3]
I171201 16:16:02.270053 52914 storage/raft_transport.go:455  raft transport stream to node 1 established
I171201 16:16:02.287873 52918 storage/replica.go:3158  [s1,r1/1:/M{in-ax}] proposing ADD_REPLICA((n3,s3):3): updated=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
------- Stdout: -------
I171201 16:14:13.439833 52168 storage/store.go:1198  [s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I171201 16:14:13.439901 52168 gossip/gossip.go:331  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:35819" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
W171201 16:14:13.450036 52168 gossip/gossip.go:1277  [n2] no incoming or outgoing connections
I171201 16:14:13.450145 52168 storage/store.go:1198  [s2] [n2,s2]: failed initial metrics computation: [n2,s2]: system config not yet available
I171201 16:14:13.450197 52168 gossip/gossip.go:331  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:33085" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
I171201 16:14:13.450281 52346 gossip/client.go:129  [n2] started gossip client to 127.0.0.1:35819
W171201 16:14:13.515283 52168 gossip/gossip.go:1277  [n3] no incoming or outgoing connections
I171201 16:14:13.515431 52168 storage/store.go:1198  [s3] [n3,s3]: failed initial metrics computation: [n3,s3]: system config not yet available
I171201 16:14:13.515494 52168 gossip/gossip.go:331  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:42183" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > 
I171201 16:14:13.515506 52530 gossip/client.go:129  [n3] started gossip client to 127.0.0.1:35819
I171201 16:14:13.517280 52596 storage/replica_raftstorage.go:527  [s1,r1/1:/M{in-ax}] generated preemptive snapshot 5a90cc93 at index 15
I171201 16:14:13.517845 52596 storage/store.go:3548  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):?: kv pairs: 34, log entries: 5, rate-limit: 2.0 MiB/sec, 0ms
I171201 16:14:13.518224 52438 storage/replica_raftstorage.go:733  [s2,r1/?:{-}] applying preemptive snapshot at index 15 (id=5a90cc93, encoded size=4720, 1 rocksdb batches, 5 log entries)
I171201 16:14:13.518973 52438 storage/replica_raftstorage.go:739  [s2,r1/?:/M{in-ax}] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I171201 16:14:13.519353 52596 storage/replica_command.go:2153  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n2,s2):2): read existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2]
I171201 16:14:13.520351 52442 storage/replica.go:3158  [s1,r1/1:/M{in-ax}] proposing ADD_REPLICA((n2,s2):2): updated=[(n1,s1):1 (n2,s2):2] next=3
I171201 16:14:13.521316 52601 storage/replica_raftstorage.go:527  [s1,r1/1:/M{in-ax}] generated preemptive snapshot f5c8cfff at index 17
I171201 16:14:13.521844 52601 storage/store.go:3548  [s1,r1/1:/M{in-ax}] streamed snapshot to (n3,s3):?: kv pairs: 37, log entries: 7, rate-limit: 2.0 MiB/sec, 0ms
I171201 16:14:13.522261 52427 storage/replica_raftstorage.go:733  [s3,r1/?:{-}] applying preemptive snapshot at index 17 (id=f5c8cfff, encoded size=5747, 1 rocksdb batches, 7 log entries)
I171201 16:14:13.523053 52427 storage/replica_raftstorage.go:739  [s3,r1/?:/M{in-ax}] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I171201 16:14:13.523409 52601 storage/replica_command.go:2153  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n3,s3):3): read existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, next=3]
I171201 16:14:13.524198 52447 storage/raft_transport.go:455  raft transport stream to node 1 established
I171201 16:14:13.525538 52617 storage/replica.go:3158  [s1,r1/1:/M{in-ax}] proposing ADD_REPLICA((n3,s3):3): updated=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
W171201 16:14:13.664615 52429 storage/raft_transport.go:281  unable to accept Raft message from (n3,s3):3: no handler registered for (n1,s1):1
W171201 16:14:13.664727 52429 storage/raft_transport.go:281  unable to accept Raft message from (n2,s2):2: no handler registered for (n1,s1):1
W171201 16:14:13.664767 52429 storage/raft_transport.go:281  unable to accept Raft message from (n2,s2):2: no handler registered for (n1,s1):1
W171201 16:14:13.664794 52429 storage/raft_transport.go:281  unable to accept Raft message from (n2,s2):2: no handler registered for (n1,s1):1
W171201 16:14:13.665063 52449 storage/raft_transport.go:494  no handler found for store 3 in response range_id:1 from_replica:<node_id:1 store_id:1 replica_id:1 > to_replica:<node_id:3 store_id:3 replica_id:3 > union:<error:<message:"store 1 was not found" transaction_restart:NONE origin_node:0 detail:<store_not_found:<store_id:1 > > now:<> > > 

```

Please assign, take a look and update the issue accordingly.

--- a/pkg/storage/client_raft_test.go
+++ b/pkg/storage/client_raft_test.go
@@ -1946,7 +1946,8 @@ func TestWedgedReplicaDetection(t *testing.T) {
 	// not past it.
 	mtc.manualClock.Increment(storage.MaxQuotaReplicaLivenessDuration.Nanoseconds() - 1)
 
-	// Send a request to the leader replica to update the quota pool status.
+	// Send a request to the leader replica. followerRepl is locked so it will
+	// not respond.
 	ctx := context.Background()
 	key := roachpb.Key("k")
 	value := []byte("value")
@@ -1961,18 +1962,26 @@ func TestWedgedReplicaDetection(t *testing.T) {
 		t.Fatalf("expected follower to still be considered active")
 	}
 
-	// Increment leader's clock past MaxQuotaReplicaLivenessDuration
-	mtc.manualClock.Increment(storage.MaxQuotaReplicaLivenessDuration.Nanoseconds() + 1)
+	// It is possible that there are in-flight heartbeat responses from
+	// followerRepl from before it was locked. The receipt of one of these
+	// would bump the last active timestamp on the leader. Because of this,
+	// we check whether the follower is eventually considered inactive.
+	testutils.SucceedsSoon(t, func() error {
+		// Increment leader's clock past MaxQuotaReplicaLivenessDuration
+		mtc.manualClock.Increment(storage.MaxQuotaReplicaLivenessDuration.Nanoseconds() + 1)
 
-	// Send another request to the leader replica to update the quota pool status.
-	if _, pErr := client.SendWrapped(ctx, leaderRepl, req); pErr != nil {
-		t.Fatal(pErr)
-	}
+		// Send another request to the leader replica. followerRepl is locked
+		// so it will not respond.
+		if _, pErr := client.SendWrapped(ctx, leaderRepl, req); pErr != nil {
+			t.Fatal(pErr)
+		}
 
-	// The follower should no longer be considered active.
-	if leaderRepl.IsFollowerActive(ctx, followerID) {
-		t.Fatalf("expected follower to be considered inactive")
-	}
+		// The follower should no longer be considered active.
+		if leaderRepl.IsFollowerActive(ctx, followerID) {
+			return errors.New("expected follower to be considered inactive")
+		}
+		return nil
+	})
 }
 
 // TestRaftHeartbeats verifies that coalesced heartbeats are correctly

